syntax = "proto3";

package conf;

message GameConf {
    int32 board_size = 1;
    int32 win_num = 2;
}

message ModelConf {
    string model_path = 1;
    int32 residual_blocks = 2;
    int32 batch_size = 3;
    float l2_c = 4;
    float lr = 5;
    string train_model_weight_save_path = 6;
    string predict_model_weight_save_path = 7;
    bool load_pre = 8;
}

message SelfPlayConf {
    int32 data_pool_size = 1;
    int32 producer_threads_num = 2;
    int32 producer_log_freq = 3;
    int32 examiner_threads_num = 4;
    int32 examiner_game_num = 5;
    int32 examiner_interval = 6;
    int32 train_step_interval = 7;
    int32 trainer_log_freq = 8;
}

message MctsConf {
    int32 simulate_num = 1;
    int32 c_puct = 2;
    int32 explore_steps = 3;
    float dirichlet_esp = 4;
    float dirichlet_alpha = 5;
}

message DataConf {
    int32 init_game_num = 1;
    int32 batch_game_num = 2;
    string data_path = 3;
}

message UiConf {
    int32 background_size = 1;
    int32 game_area = 2;
    string background = 3;
}

message Conf {
    GameConf game_conf = 1;
    ModelConf model_conf = 2;
    MctsConf mtcs_conf = 3;
    DataConf data_conf = 4;
    UiConf ui_conf = 5;
    SelfPlayConf self_play_conf = 6;
}

